common/utils.py

```diff
@@ -52,7 +60,36 @@ def get_date(day_arg):
     return date.today() + timedelta(days=day_num)
 
 
-def render(filename, context={}, path='templates'):
+def date_to_str(day_arg):
+    """
+    This function converts the python date format gotten from get_date function
+    to string for graphql api request.
+    """
+    date = get_date(day_arg)
+    return date.strftime('%Y-%m-%d')
+
+
+def render(filename, context={}, error=None, path='templates'):
+    if error:
+        context['error'] = error
     return jinja2.Environment(
         loader=jinja2.FileSystemLoader(path)
     ).get_template(filename).render(context)
+
+
+def make_api_request(query):
+    api_url = dotenv.get('API_URL_ENDPOINT')
+    endpoint = '{}?query={}'.format(api_url, query)
+    headers = {'X-TavernaToken': dotenv.get('X-TAVERNATOKEN')}
+    return requests.post(endpoint, headers=headers).json()
+
+
+def make_api_request_for_timetables():
+    query = 'query{timetables{edges{node{name, slug, cycleLength,refCycleDay, \
+             vendors{edges{node{name}}}, admins{edges{node{username}}}}}}}'
+    return make_api_request(query)['timetables']
```

Comment:
No need, because of the way our queries are structured in GraphQL. Please ignore